app/src/main/java/com/example/tinder/MainActivity.java

```diff
@@ -92,12 +93,16 @@ private void initLocation() {
     private void checkLogin() {
         // check login
         User user = User.getLocalUser(getPreferences(MODE_PRIVATE));
+        // first time login
         if (user == null) {
             userAuth.setState(UserAuth.UN_AUTHENTICATED, UserAuth.NONE);
             Log.d("authen", "continue");
         } else {
+            // get information from c
             userAuth.setState(UserAuth.AUTHENTICATED, UserAuth.NONE);
             userAuth.setUser(user);
+            String authenToken = user.getAuthen_token();
+            userAuth.setSocketIO(new SocketIO("http://167.99.69.92:8889", authenToken));
```

Comment:
đừng set SocketIO ở đây

Reply:
T chỉ tạo socket 1 lần thôi, rồi xài chung cho toàn bôh

Reply:
biết rồi, ý là m chỉ cần gọi 1 lần ở switch case: UserAuth.AUTHENTICATED, ngay chổ addControls của MainActivity. Vì bất cứ khi nào đăng nhập thành công đều nhảy vào chổ đó.